Userland/Libraries/LibWeb/Painting/BorderPainting.cpp

```diff
@@ -137,7 +138,66 @@ void paint_border(PaintContext& context, BorderEdge edge, Gfx::IntRect const& re
         return;
     }
 
-    context.painter().fill_rect(rect, color);
+    auto draw_horizontal_or_vertical_line = [&](auto p1, auto p2) {
+        // Note: Using fill_rect() here since draw_line() produces some overlapping pixels
+        // at the end of a line, which cause issues on borders with transparency.
+        p2.translate_by(1, 1);
+        context.painter().fill_rect(Gfx::IntRect::from_two_points(p1, p2), color);
+    };
+
+    auto draw_border = [&](auto const& border, auto const& radius, auto const& opposite_border, auto const& opposite_radius, auto p1_step_translate, auto p2_step_translate) {
+        auto current_p1 = p1.to_type<float>();
+        auto current_p2 = p2.to_type<float>();
+        auto p1_step = radius ? 0 : border.width / static_cast<float>(int_width);
+        auto p2_step = opposite_radius ? 0 : opposite_border.width / static_cast<float>(int_width);
+        for (int i = 0; i < int_width; ++i) {
+            draw_horizontal_or_vertical_line(current_p1.to_type<int>(), current_p2.to_type<int>());
+            p1_step_translate(current_p1, p1_step);
+            p2_step_translate(current_p2, p2_step);
+        }
+    };
+
+    // FIXME: There is some overlap where two borders (without border radii meet),
+    // which produces artifacts if the border color has some transparency.
+    // (this only happens if the angle between the two borders is not 45 degrees)
+    switch (edge) {
+    case BorderEdge::Top:
+        return draw_border(
```

Comment:
`draw_border()` doesn't return anything, so this is confusing. Just `break` and implicit return at the end of the function please.